File: sem iv/Theory of Computaion/lab/code/c/2_substring_count.cpp
// Program: Extract substring given starting position (a) and number of symbols (b)
#include <iostream>
#include <cstring>
using namespace std;

void substring2(char w[100], int start, int noOfSymbol);

int main() {
    char input[100];
    int start, noOfSymbol;

    cout << "Enter the string" << endl;
    cin.getline(input, 100);

    do {
        cout << "Enter the starting position of the substring" << endl;
        cin >> start;
        cout << "Enter the number of symbol to be extracted" << endl;
        cin >> noOfSymbol;
    } while (start < 1 || noOfSymbol > (int)strlen(input));

    substring2(input, start - 1, noOfSymbol);
    cin.ignore();
    cin.get();
    return 0;
}

void substring2(char w[100], int start, int noOfSymbol) {
    cout << "The substring is" << endl;
    for (int i = start, count = 0; count < noOfSymbol && w[i] != '\0'; i++, count++)
        cout << w[i];
    cout << endl;
}

File: sem iv/Theory of Computaion/lab/code/python/2_substring_count.py
# Program: Extract substring given starting position (a) and number of symbols (b)

def substring2(w, start, no_of_symbol):
    print("The substring is")
    print(w[start:start + no_of_symbol])

def main():
    inp = input("Enter the string\n")
    while True:
        start = int(input("Enter the starting position of the substring\n"))
        no_of_symbol = int(input("Enter the number of symbol to be extracted\n"))
        if not (start < 1 or no_of_symbol > len(inp)):
            break
    substring2(inp, start - 1, no_of_symbol)

if __name__ == "__main__":
    main()
